.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  book-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install mdbook
+        run: |
+          cargo install --version "^0.4.5" mdbook
+          cargo install --version "^1.3.1" skill-tree
+
+      - name: mdbook test
+        run: |
+          mdbook build
+          mdbook test
+
+      - name: check skill-tree
+        run: skill-tree roadmap.toml book/roadmap
+
+      - name: Deploy GitHub Pages
+        if: github.ref == 'refs/heads/master'
+        run: |
+          git worktree add gh-pages gh-pages
+          git config user.name "Deploy from CI"
+          git config user.email ""
+          cd gh-pages
+          # Delete the ref to avoid keeping history.
+          git update-ref -d refs/heads/gh-pages
+          rm -rf *
+          mv ../book/* .
+          git add .
+          git commit -m "Deploy $GITHUB_SHA to gh-pages"
+          git push --force
```

Comment:
is this copied from somewhere? There's a lot git munging here, are you sure it works?

Reply:
Yeah, I use it on libc crate: https://github.com/rust-lang/libc/blob/master/.github/workflows/docs.yml
And as I said on OP, it works fine on my fork: https://johntitor.github.io/types-team/